lib/puppet/util.rb

```diff
@@ -25,6 +25,67 @@ module Util
 
   extend Puppet::Util::SymbolicFileMode
 
+  public
+  def self.get_environment_variable(name, mode = nil)
+    if mode.nil?
+      if Puppet.features.microsoft_windows?
+        mode = :windows
+      else
+        mode = :posix
+      end
+    end
+
+    if mode == :windows
+      Puppet::Util::Windows::Process.get_environment_strings.each do |key, value |
+        if name.casecmp(key) == 0 then
+          return value
+        end
+      end
+      return nil
+    else
+      ENV[name]
+    end
+  end
+  # TODO use alias_method instead
+  def self.env(name,mode = nil)
```

Comment:
While there are some appealing reasons to patch `ENV`, I'd definitely prefer to find another way.  I think the simplest thing that works here is to just change `self.env` to `self.get_env`, and add a note about the Ruby bug / that the code can be removed when we're on 2.3.